### DISCUSSION TITLE
Can you test this one?

### DISCUSSION THREAD

**Discussion by rmast** [rmast/djvulibre#1] [Announcements]
@jsbien I pushed a new commit with a feasible csv line format onto the revealshapes-branch:
sjbz or djbz,page number,blit number,blit shapeno,shape bits columns,rows,rowsize,blit bottom, left

**Comment by jsbien** [comment]
Great! But please remind me how to compile it. I'm on Debian stable (bookworm).

**Comment by rmast** [comment]
Just follow the regular compile instructions for djvulibre. Then the revealshapes executable appears in the tools-directory. I guess as exportshapes is also in that branch, that the mysql-dependencies are also required.

Verzonden vanaf Outlook voor Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by jsbien** [reply]
From README:
> Chances are that everything will work by simply running 
commands "./configure", "make" and "make install".

But I forgot how to create `configure` :-( I circumvented the problem by calling `dpkg-buildpackage -uc -us -b -rfakeroot`. I got

`Emergency save document locations:`
 `/home/jsbien/git/revealshapes/desktopfiles/djvu.svg.2024_04_07_09_35_35.0.svg`
`Emergency save completed. Inkscape will close now.`
`If you can reproduce this crash, please file a bug at https://inkscape.org/report`
`with a detailed description of the steps leading to the crash, so we can fix it.`

It was followed by
`make[2]: *** [Makefile:597: 16x16/mimetypes/image-vnd.djvu.png] Segmentation fault`

Moreover when I tried to open the svg file I got
![Screenshot from 2024-04-07 09-44-35](https://github.com/rmast/revealshapes/assets/8024702/9f9bfb7e-4a47-4329-88c8-52f5d3c1d614)

**Reply by rmast** [reply]
I am now building djvulibre master on Mint 20 with inkscape, and it seems to build just fine with inkscape of Mint 20. Unless you had a special version of inkscape installed the bookworm build of djvulibre probably breaks on having inkscape installed.

**Reply by rmast** [reply]
The same construction with Magick convert also does build on Mint 20. So it probably broke in more recent versions. When I was running the ubuntu-latest action (Ubuntu 22.04) this conversion in djvulibre master was already broken.

**Reply by rmast** [reply]
I was finally able to find the right correction of the github-action to compile the conversion with Magick on the ubuntu-latest image:
sudo apt install -y libfreetype6-dev gsfonts libmagickwand-dev imagemagick
It appears to also solve the Inkscape issue.

**Reply by jsbien** [reply]
I don't need Inkscape for the time being, so I will investigate this later.

**Reply by rmast** [reply]
I installed inkscape on my bookworm VM. The build just ran fine with inkscape conversion, even without the trick I had to do on the git action. The issue is not quite clear. I didn't start bookworm from the fully fledged desktop, but from netinstall with XFCE4+SSH.

**Comment by rmast** [comment]
This afternoon around 15:00 CEST I will have time to set up a teams-call or so.

Verzonden vanaf Outlook voor Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by jsbien** [comment]
Do I understand correctly you want to contact me with Microsoft Teams? I have it installed, but used only once and with some problems. I'm registered there as jsbien@uw.edu.pl (not jsbien@mimuw.edu.pl).

**Comment by rmast** [comment]
By the way, Debian Bookworm is shiny new. Usually that introduces all kinds of dependency-issues. I myself am still on a 4 year old distro. I can try to make a GitHub-action for compiling on Bookworm.

Verzonden vanaf Outlook voor Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by jsbien** [reply]
I looked for a development library for Inscape, but there seem to be no such package...

**Comment by rmast** [comment]
If Teams is running in debian then there will certainly be some hoops to take, but screen sharing is possible. My phone allows calling Poland if audio is an issue.

Verzonden vanaf Outlook voor Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by jsbien** [reply]
I hope calling me by phone won't be necessary. My problem could be audio, but it was in 2021 and there were several substantial upgrades from that time. Looks like it is not possible to check it in advance?

BTW, do you use Zoom?

**Comment by rmast** [comment]
I can use Zoom as well, but never sent an invitation. I sent you a Teams-invitation, if you visit the link it allows testing in advance. If you prefer Zoom you can reply that message with a Zoom-link.

Verzonden vanaf Outlook voor Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by jsbien** [reply]
As for Zoom, it's the same for me: I use it quite often, but never sent and invitation. 

I have to leave home for an hour or so, then I will use your link to test things in advance. If I encounter serious problems, I will try to send the Zoom invitation.

**Comment by jsbien** [comment]
Just in case I sent you a Zoom invitation (at least I think so....) for 15.30.

**Comment by rmast** [comment]
When you get Inkscape out of the way, these were about the things I had to install on bookworm to make revealshapes compile:
apt install git build-essential autopoint automake libtool libguestfs-tools autotools-dev libmariadb-dev-compat libmariadb-dev autoconf-archive libboost-dev

**Reply by jsbien** [reply]
Most of them were already installed.

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1068585

**Reply by rmast** [reply]
That desktopfiles directory contains a reference to inkscape:
https://github.com/rmast/djvulibre/blob/main/desktopfiles/Makefile.am
```
if HAVE_CONVERSION_INKSCAPE
convert_icons_process = \
s=`echo $@ | sed -e 's/[a-z]*\([0-9]*\).*/\1/'`; \
${INKSCAPE} \
--without-gui \
--export-width=$${s} \
--export-height=$${s} \
--export-png=$@ $<
endif

In the Github action pipeline it tries to use convert instead and ends up in another error, which seems related to quotes ` ' :
convert: unable to read font `helvetica' @ error/annotate.c/RenderFreetype/1396.
convert: non-conforming drawing primitive definition `text' @ 
```
If none of those is installed it skips this conversion-test.
